### PR TITLE
batching for aggregated data requests

### PR DESCRIPTION
**Pull request by boweihan**
## Overview
Batch requests by frame for aggregated property data.

Screen Shot 2022-05-24 at 4 30 47 PM

Implementation re-uses elements from historical batching - https://github.com/awslabs/iot-app-kit/pull/102.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -38,6 +38,7 @@ __diff_output__
 # Cypress screenshots
 **/cypress/screenshots
 **/cypress/videos
+**/cypress/snapshots/All Specs
```
Comment (square-li) [inline .gitignore:41]: Do we need to escape the space character in the path? Like `".../All\ Spacs"`

Comment (boweihan) [reply .gitignore:41]: I believe this works, at least it works on my mac.

https://git-scm.com/docs/gitignore

gitignores are line-based and it looks like only trailing spaces need to be escaped.

Comment (square-li) [reply .gitignore:41]: I am more worried about Windows. I am going to verify it. But everything else looks good!

Comment (square-li) [reply .gitignore:41]: I just verified that it works for windows as well.

**Review by square-li — APPROVED** [review]

**Pull request merged**